pbs.py

```diff
@@ -1,136 +1,148 @@
-import variables
 import re
+import sys
+import os
+import variables
 
-def make_pbsnodes_yaml(fin, fout):
+
+def make_pbsnodes_yaml(orig_file, yaml_file):
     """
     read PBSNODES_ORIG_FILE sequentially and put in respective yaml file
     """
-    OfflineDownNodes = 0 
-    # global OfflineDownNodes
-
-    for line in fin:
-        line.strip()
-        searchdname = '^\w+([.-]?\w+)*'
-        if re.search(searchdname, line) is not None:   # line containing domain name
-            m = re.search(searchdname, line)
-            dname = m.group(0)
-            fout.write('domainname: ' + dname + '\n')
+    if not os.path.getsize(orig_file) > 0:
+        print 'Bailing out... Not yet ready for Sun Grid Engine clusters'
+        # os.chdir(HOMEPATH + 'qt')
+        sys.exit(0)
 
-        elif 'state = ' in line:
-            nextchar = line.split()[2][0]
-            if nextchar == 'f':
-                state = '-'
-            elif (nextchar == 'd') | (nextchar == 'o'):
-                state = nextchar
-                OfflineDownNodes += 1
-            else:
-                state = nextchar
-            fout.write('state: ' + state + '\n')
+    with open(orig_file, 'r') as fin, open(yaml_file, 'a') as fout:
+        for line in fin:
+            line.strip()
+            search_domain_name = '^\w+([.-]?\w+)*'
+            if re.search(search_domain_name, line) is not None:   # line containing domain name
+                m = re.search(search_domain_name, line)
+                domain_name = m.group(0)
+                fout.write('domainname: ' + domain_name + '\n')
 
-        elif 'np = ' in line or 'pcpus = ' in line:
-            np = line.split()[2][0:]
-            # TotalCores = int(np)
-            fout.write('np: ' + np + '\n')
+            elif 'state = ' in line:
+                nextchar = line.split()[2][0]
+                if nextchar == 'f':
+                    state = '-'
+                elif (nextchar == 'd') | (nextchar == 'o'):

```

Comment:
if/elif/elif structures are often optimized via table structures... is it possible also here?